### PR TITLE
chore(repo): add documentation and clean up stories

### LINKED ISSUE #771: [Storybook] Re-evaluate documentation of components with >10 stories
### Summary

Each story for a component is intended to represent [a single visual state of that component](https://storybook.js.org/docs/basics/writing-stories/). With complex components (`Table`) or components with many variants (`TimeSeriesCard`, `ValueCard`, etc.), the growing story list is difficult to navigate.

Atop this cognitive load, we also have `Stateful` stories that provide a stateful (interactive) example of the component. Some have multiple stateful examples, further clouding understanding.

#### Some ideas to remedy:
- [ ] Add documentation to the welcome story restating storybook's intention - each story represents a single visual state of a component, noting the exception of `Stateful` interactive stories. This distinction of `static` vs `interactive` is currently not clear for users not familiar with storybook.
- [ ] Build a list of crucial/important/essential stories for each component with >10 stories. This core set can still be more than 10, we're just trying to cut back from huge numbers like 30 😳 .
- [ ] Refactor to remove the non-essential stories - replace these with well crafted `enzyme`/`@testing-library/react` tests looking at props, event handlers, etc. to cover the functionality previously covered by storyshots.
- [ ] If a component has multiple stateful stories, combine these together into one stateful component with the most common use case. We may be able to enable further configuration via knobs to turn off different feature sets within a component.
- [ ] Standardize story names. A common format we could follow is "with {feature}". Stateful stories should always have the same format. Commonly we've used "Stateful{Component} Example", minor point of clarity could be an additional space "Stateful {Component} Example"
- [ ] Improve documentation of complex components via a readme for that component. We may find a link to a codesandbox.io helpful to provide a playground environment.

Are there other ways we could improve this?

<sub>[Internal tracking issue 81](https://github.ibm.com/Watson-IoT/pal-tracking/issues/81)</sub>

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #771 

**Summary**

- Reduced the number of stories for components that have more than 10 stories by adding more knobs to allow consumers to change the props, add tests to cover simple things like i18n, and increased documentation.
- The table has been excluded from this, but a new issue #2405 has been added to tackle the table. Blocked by #2336 currently (6/14/2021).

**Change List (commits, features, bugs, etc)**

- BarChartCard: combine stories with knobs and added docs on Readme
- Card: combined stories with knobs and added docs on Readme
- ComposedModal: combined stories with knobs and added docs on Readme
- List: added docs to Readme tab
- SimpleList: combined stories with knobs and added docs on Readme
- PageTitleBar: combined stories with knobs and added docs on Readme
- TableCard: combined stories with knobs, added docs on Readme, unit tests, fixed missing proptypes
- TimeSeriesCard: combined stories with knobs and added docs on Readme
- ValueCard: combined stories with knobs and added docs on Readme

**Acceptance Test (how to verify the PR)**

- Check the stories still achieve their purpose and should be more clear how different props interact now that they can be controlled via knobs.
- The documentation shows props for these components

**Regression Test (how to make sure this PR doesn't break old functionality)**

- stories still function as expected

### CODE REVIEW

File: packages/react/src/components/List/ListItem/_list-item.scss [hunk 1]
```diff
@@ -93,7 +93,7 @@
   color: $text-02;
   padding: $spacing-03 $spacing-05;
   align-items: center;
-  overflow-x: hidden;
```
Comment (davidicus) [inline packages/react/src/components/List/ListItem/_list-item.scss:96]: I see two actual code file changed with this PR. Just making sure it is intentional

Comment (kevinsperrine) [reply packages/react/src/components/List/ListItem/_list-item.scss:96]: @davidicus For this one, yes, It was a bug I noticed as I was going through the stories. See next where the overflow is triggering a scroll. https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-listitem--with-overflow-menu-row-actions

**Review by davidicus — APPROVED** [review]
Nice work

**Comment by sstone2423** [comment]
The netlify preview is broken for me unfortunately. Getting these warnings but not sure if they're related

Screen Shot 2021-06-15 at 9 22 16 AM

**Comment by kevinsperrine** [comment]
> The netlify preview is broken for me unfortunately. Getting these warnings but not sure if they're related
> 
> Screen Shot 2021-06-15 at 9 22 16 AM

@sstone2423 The netlify preview links have started appending `iframe` to the link for some reason. 🤷‍♂️  It's working without it though. https://deploy-preview-2359--carbon-addons-iot-react.netlify.app/

**Comment by bjornalm** [comment]
Regarding the naming of the readme files. I see that this PR uses the name of the component, i.e. BarChartCard.md whereas the existing component readme files are name all README.md. I don't care which naming scheme we use but I think we should be consistent.

**Comment by kevinsperrine** [comment]
> Regarding the naming of the readme files. I see that this PR uses the name of the component, i.e. BarChartCard.md whereas the existing component readme files are name all README.md. I don't care which naming scheme we use but I think we should be consistent.

Yeah, I opted for the component name for fuzzy search reasons. If you search for README in vscode, you'd still have to add part of the component name to narrow it down. I'm fine with either, too.

File: packages/react/src/components/PageTitleBar/PageTitleBar.story.jsx [hunk 2]
```diff
@@ -145,13 +131,13 @@ export const WithStatusDescriptionAndCrumbs = () => (
...
   </div>
-);
+));
 
 WithStatusDescriptionAndCrumbs.story = {
   name: 'with status description and breadcrumbs',
 };
 
-export const WithEditableTitleBar = () => (
+export const WithEditableTitleBar = withReadme(README, () => (
```
Comment (bjornalm) [inline packages/react/src/components/PageTitleBar/PageTitleBar.story.jsx:140]: This one is a knob in the other stories, BUT the action callback is not working. Probably related to the wrapper function again. We should try to find the root cause of this...

**Review by bjornalm — CHANGES_REQUESTED** [review]

**Comment by bjornalm** [comment]
Really nice work! I understand this is a lot.  I haven't gone through all the changes, but except for typos I think several individual remarks are applicable to more components in the PR:

* Consistency when describing complex types
* Consistency regarding existing "Getting started" content 
* When splitting objects like i18n over multiple rows we should add the defaults to the rows
* Knobs are also their own stories
* Story actions not being called (weird function wrapper problem)

File: packages/react/src/components/ComposedModal/ComposedModal.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+| footer.secondaryButtonLabel                    | string                                      |                 |                                                                                                                                                                              |
+| footer.isPrimaryButtonHidden                   | bool                                        |                 | should the primary button be hidden (i.e. only show Cancel)                                                                                                                  |
+| footer.isPrimaryButtonDisabled                 | bool                                        |                 | should the primary button be disabled                                                                                                                                        |
+| type                                           | enum:<br>&nbsp;'warn'<br>&nbsp;'normal'<br> | null            | NEW PROP: Type of dialog, affects colors, styles of dialog                                                                                                                   |
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/ComposedModal/ComposedModal.md:106]: It was in the comment describing the prop, so it got pulled in from that. I'll remove it.

File: packages/react/src/components/BarChartCard/BarChartCard.md [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+| i18n                                  | shape                                                                                                                     | { noDataLabel: 'No data', tooltipGroupLabel: 'Group', tooltipTotalLabel: 'Total', defaultFilterStringPlaceholdText: 'Filter',}                                                 | internationalization                                                                                                                                                                                                                                                                                                                             |
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/BarChartCard/BarChartCard.md:314]: agreed.

File: packages/react/src/components/BarChartCard/BarChartCard.md [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+- [Simple bar chart](#simple-bar-chart)
+- [Grouped bar chart](#grouped-bar-chart)
+- [Props](#props)
+- [External links](#external-links)
+  - [Source Code](#source-code)
+  - [Feedback](#feedback)
+
+## Getting Started
+
+The `BarChartCard` wraps a Card display around @carbon/charts to render simple bar charts, grouped bar charts, or stacked bar charts.
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/BarChartCard/BarChartCard.md:15]: agreed.

File: packages/react/src/components/List/List.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+| pagination.totalItemsText  | string                                                                                                                                          |                                                                              |                                                                                    |
+| pagination.testID          | string                                                                                                                                          |                                                                              |                                                                                    |
+| expandedIds                | array\<string\>                                                                                                                                 | []                                                                           | ids of row expanded                                                                |
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/List/List.md:56]: I'll rewrite them all to use `arrayOf(x)`.

File: packages/react/src/components/List/List.story.jsx [hunk 7]
```diff
@@ -611,13 +613,13 @@ export const WithTags = () => (
...
   </div>
-);
+));
 
 WithTags.story = {
   name: 'with tags',
 };
 
-export const IsLoading = () => (
+export const IsLoading = withReadme(README, () => (
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/List/List.story.jsx:622]: removed.

File: packages/react/src/components/BarChartCard/BarChartCard.md [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+| domainRange                           | array<number\Date>                                                                                                        | null                                                                                                                                                                           | optional domain to graph from. First value is the beginning of the range. Second value is the end of the range can be date instance or timestamp                                                                                                                                                                                                 |
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/BarChartCard/BarChartCard.md:347]: I'll rewrite to use the `arrayOf(x)` syntax

File: packages/react/src/components/BarChartCard/BarChartCard.story.jsx [hunk 9]
```diff
@@ -1,565 +1,353 @@
...
-    <div style={{ width: `${getCardMinSize('lg', size).x}px`, margin: 20 }}>
+    <div style={{ width: `${getCardMinSize(breakpoint, size).x}px`, margin: 20 }}>
       <BarChartCard
         title={text('title', 'Particles by city')}
-        id="simple-horizontal-sample"
+        id="simple-sample"
+        breakpoint={breakpoint}
         isLoading={boolean('isLoading', false)}
         isEditable={boolean('isEditable', false)}
         isExpanded={boolean('isExpandable', false)}
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/BarChartCard/BarChartCard.story.jsx:40]: fixed.

File: packages/react/src/components/TableCard/TableCard.md [hunk 10]
```diff
@@ -0,0 +1,373 @@
...
+## Passing Variables
+
+Passing variables
+To pass a variable into your card, identify a variable to be used by wrapping it in curly brackets.
+Make sure you have added a prop called 'cardVariables' to your card that is an object with key value pairs such that the key is the variable name and the value is the value to replace it with.
+Optionally you may use a callback as the cardVariables value that will be given the variable and the card as arguments.
+Note: if using row-specific variables in a TableCard href (ie a variable that has a different value per row),
+do NOT pass the cardVariables prop and be sure that your table has reference to the proper value in another column
+
+````jsx
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/TableCard/TableCard.md:98]: nice catch.

File: packages/react/src/components/TableCard/TableCard.md [hunk 11]
```diff
@@ -0,0 +1,373 @@
...
+      size={size}
+      isLoading={boolean('isLoading', false)}
+    />
+  </div>
+);
+```
+
+## Passing Variables
+
+Passing variables
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/TableCard/TableCard.md:91]: removed.

File: packages/react/src/components/BarChartCard/BarChartCard.md [hunk 12]
```diff
@@ -0,0 +1,360 @@
...
+| cardDimensions                        | shape                                                                                                                     |                                                                                                                                                                                | array of configurable sizes to dimensions                                                                                                                                                                                                                                                                                                        |
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/BarChartCard/BarChartCard.md:192]: Agreed. I'll copy them out into their own blocks. This boils down to boredom after looking at this for a while. At this point, I was basically just copy/pasting those shared sections of the cards. 🥴

File: packages/react/src/components/PageTitleBar/PageTitleBar.story.jsx (hunk 2 above)
Comment (kevinsperrine) [reply packages/react/src/components/PageTitleBar/PageTitleBar.story.jsx:140]: Yeah, this is the exact same thing as the `onToggleAggregations` callback or button `onClick` callback. It works locally and in next, but fails in these storybook netlify deploys. I think it has to be related to netlify, but it even fails if you're not opening it within the storybook frame, but opening the [iframe directly](https://deploy-preview-2359--carbon-addons-iot-react.netlify.app/iframe.html?id=1-watson-iot-pagetitlebar--with-editable-title-bar&args=). 🤷‍♂️

File: packages/react/src/components/Card/Card.md [hunk 13]
```diff
@@ -0,0 +1,656 @@
...
+- [Props](#props)
+- [External links](#external-links)
+  - [Source Code](#source-code)
+  - [Feedback](#feedback)
+
+## Getting Started
+
+The `Card` component is the foundation of all cards in the library.
+
+## Custom Card Content
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/Card/Card.md]: I copied it out of a story but never finished documenting it. It was an example of implementing a custom card.

**Comment by kevinsperrine** [comment]
> Really nice work! I understand this is a lot. I haven't gone through all the changes, but except for typos I think several individual remarks are applicable to more components in the PR:
> 
> * Consistency when describing complex types

opted for the `arrayOf(x)` description

> * Consistency regarding existing "Getting started" content

add simple description and `import` statements to the getting started section

> * When splitting objects like i18n over multiple rows we should add the defaults to the rows

filled in i18n defaults for cards on their own lines.

> * Knobs are also their own stories

removed those duplicates.

> * Story actions not being called (weird function wrapper problem)

no idea here.

**Comment by bjornalm** [comment]
> > Regarding the naming of the readme files. I see that this PR uses the name of the component, i.e. BarChartCard.md whereas the existing component readme files are name all README.md. I don't care which naming scheme we use but I think we should be consistent.
> 
> Yeah, I opted for the component name for fuzzy search reasons. If you search for README in vscode, you'd still have to add part of the component name to narrow it down. I'm fine with either, too.

Cool, so if you stick with the component name could you create an Issue to modify the naming  of the other readme files?

**Review by bjornalm — APPROVED** [review]
Good work! 
On my end I'll update https://github.com/carbon-design-system/carbon-addons-iot-react/pull/2406 to use  "Getting started" section with import. Let's take a decision today on the naming of the readme files.

**Comment by kevinsperrine** [comment]
> Good work!
> On my end I'll update #2406 to use "Getting started" section with import. Let's take a decision today on the naming of the readme files.

Let's stick with {COMPONENT_NAME}.md. I'll open an issue for the others.

**Review by davidicus — APPROVED** [review]

**Pull request merged**